lib/fs/basicfs_unix.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2016 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// +build !windows
+
+package fs
+
+import (
+	"os"
+	"runtime"
+)
+
+var symlinksSupported = true
+
+func DisableSymlinks() {
+	symlinksSupported = false
+}
+
+func (BasicFilesystem) SymlinksSupported() bool {
+	return symlinksSupported
+}
+
+func (f *BasicFilesystem) CreateSymlink(name, target string) error {
+	name, err := f.rooted(name)
+	if err != nil {
+		return err
+	}
+	return os.Symlink(target, name)
+}
+
+func (f *BasicFilesystem) ReadSymlink(name string) (string, error) {
+	name, err := f.rooted(name)
+	if err != nil {
+		return "", err
+	}
+	return os.Readlink(name)
+}
+
+func (f *BasicFilesystem) MkdirAll(name string, perm FileMode) error {
+	name, err := f.rooted(name)
+	if err != nil {
+		return err
+	}
+	return os.MkdirAll(name, os.FileMode(perm))
+}
+
+func (f *BasicFilesystem) Show(name string) error {
+	_, err := f.rooted(name)
```

Comment:
Hiding stuff on linux is noop (apart from trying to escape the root, in which case it errors).
Only way to hide is to prefix it with a dot (which changes the name so we don't want to do that).